### ISSUE TITLE
License clause is questionable

### ISSUE THREAD

**Issue by toulouse** [skadistats/clarity#34]
long post but the gist is that this clause should be removed. At best it's worthless. At worst, it has negative value to contributors.

> 4\. Redistributions of any form whatsoever must retain the following
    acknowledgment: 'This product includes software developed by
    skadistats (http://www.skadistats.com/).' Web applications and
    properties built using the clarity library must additionally include
    a visible, legible hypertext link with text "Powered by skadistats"
    linking to the URL http://www.skadistats.com on every page containing
    data obtained by using this library. Removal of this attribution
    shall be granted immediately (within one business day) if
    requested by skadistats or its authorized agents.

This clause is incredibly questionable and appears to have been written with no real understanding of either copyright or law. For one thing, it means that the software is not in actuality open-source, because you're implicitly claiming some form of copyright/sublicensing of data. Which is nonsense. That data came from Valve and if you really wanted to fight it you'd probably get slammed for violating the Steam Subscriber Agreement anyways.

"Web applications" isn't defined. "Properties" isn't defined. "Pages" isn't defined. "Skadistats" isn't defined. Is Skadistats incorporated? It surely isn't, because it's never referred to by a legal company name. And who's *actually* authorized to make those requests? If someone fakes being a partner of skadistats and requests a site that uses the library to remove the attribution within one day, that site is required to remove it if they don't hear back, except they're not penalized either way by either attributing or not attributing skadistats. But if they were, this clause would fuck them over the instant some phisher decided to screw with them.

And what about providing the software as-is? That stuff is standard, but if you disclaim liability for that then how can the clause that implies (but doesn't say anything further) that skadistats can make a request to users of the library even be enforceable without skadistats actually *being* liable? *Doesn't that make the clause an actual risk to skadistats?*

Also, there's an obvious workaround that would allow a site to use clarity with exactly zero attribution whatsoever, assuming they're willing to argue that you have no right to sublicense Valve's copyrighted data anyways, which you don't, which the attribution requirement basically says. That is, that's what someone might do if the penalties weren't completely toothless anyways. And if they weren't, normal users of it would be vulnerable to and exploitable by griefers anyhow.

tl;dr: Penalties: toothless. Protections to you: zero or possibly negative (risky). Open-source: hardly.

**Comment by mugsy-mcgeee** [comment]
So what your saying is you would like to use the library we wrote without attribution?

**Comment by toulouse** [comment]
No, I'm not saying anyone who want to could anyways and your license would do nothing. And you happily get to be liable, potentially, if someone who isn't you decides to impersonate you, because one of your clauses is not like the others. And your license is basically not valid because it claims rights you don't have, because of your fourth clause.

**Comment by mugsy-mcgeee** [comment]
Thank you for the clarification in your stance. I likely mis-interpreted the tone of your initial post. This sounds like something we should discuss further and decide whether it makes sense to alter the license.

**Comment by mugsy-mcgeee** [comment]
Changed the issue title to be less inflammatory

**Comment by onethirtyfive** [comment]
I think we might have found the only lawyer on the Internet, guys.

But seriously, he has a point. That clause is kind of like a piracy warning on a DVD. What we didn't want was to keep it closed source, and what we didn't want moreover was to encourage people to take it without considering that a lot of hard work went into it, for which there is AMPLE precedent in this very scene.

Truth be told you could say the same thing if we had gone GPL instead. What, is Richard Stallman going to descend in his FSF chariot (YOU CANT RUN FROM HEAVEN) and personally fight for Freeasinspeech dota replay parsing in the 9th circuit?

The lawyer has a point though, it's toothless either way. One just looks worse.

**Comment by spheenik** [comment]
> YOU CANT RUN FROM HEAVEN

Damn Zeus pickers :)

On the subject though: I think the clause should be removed too. Let's ask for attribution in the docs. Nice people will do it without hesitation, because they are thankful, and the others would probably not mentioned it even with the clause in the license.
